### ISSUE TITLE
where to config the underline theme?

### ISSUE THREAD

**Issue by chunyang-wen** [luochen1990/rainbow#173]
I am using rainbow with the default configuration. That means: install rainbow with vim-plug and set it 

```vim
let g:rainbow_active = 1
```

![image](https://user-images.githubusercontent.com/2911134/141754778-dba9a44e-1a43-4076-8aa0-0eaa4712b2f2.png)

```python
aaa = 1
b = (aaa,)
```

I want to remove the underline, as it seems that it is not correct.

![image](https://user-images.githubusercontent.com/2911134/141755013-47109d59-17c6-48c4-8ff9-03a2f3baef73.png)

When I call `:RainbowToggle`, the underline is gone.

I have search google and try to read the vimscript of rainbow, but I found nothing useful.

+ `operators` config?
+ underline not found
+ cterm: no suspicious clue found

**Comment by luochen1990** [comment]
Maybe you can find something useful in README.md of this repo  (search `guis` and `cterms`), and from vim help (`:h attr-list`)

**Comment by chunyang-wen** [comment]
> 

I‘d like to remove, not add attribute. I have search *.vim in rainbow, there seems no underline configuration.

**Comment by luochen1990** [comment]
You can read the README about troubleshooting, and use the tool to check which syntax rule is actived.
Could that be spell checking?

**Comment by chunyang-wen** [comment]
Yes, it is the spell check. But I found that the configuration is different with the README.

```vim
'parentheses_options': 'contains=@Spell'
```
to turn off the spell checker.

I have add the spell related configuration in vimrc

```vim
set spell spelllang=en_us
hi clear SpellBad
hi SpellBad cterm=underline
hi clear SpellRare
hi SpellRare cterm=underline
hi clear SpellCap
hi SpellCap cterm=underline
hi clear SpellLocal
hi SpellLocal cterm=underline
```

**Comment by 91khr** [comment]
> to turn off the spell checker

But the configuration to explicitly turn off spell check in parentheses is `'parentheses_options': 'contains=@NoSpell'`. (
And by default rainbow wouldn't modify the spell check option, that is, if you don't turn on the spell check explicitly, rainbow wouldn't turn it on for you.

**Comment by CaiJinKen** [comment]
I have same question. How to cancel spell check in the plugin
vimrc:
```vim
let g:rainbow_active = 1
nmap <leader>p :RainbowToggle<cr> 
```
result: 
<img width="937" alt="image" src="https://github.com/user-attachments/assets/3877101d-b05b-4a2a-bc42-6dd2ae0ec838">

**Comment by luochen1990** [comment]
> > to turn off the spell checker
> 
> But the configuration to explicitly turn off spell check in parentheses is `'parentheses_options': 'contains=@NoSpell'`. ( And by default rainbow wouldn't modify the spell check option, that is, if you don't turn on the spell check explicitly, rainbow wouldn't turn it on for you.

The trick is ,  vim, by default, will  check if there is any syntax rules about spell checking, if there is `@Spell`, then top level default will become no spell checking, and if there is  `@NoSpell`,  the default rule will become spell checking enabled...

So the solution is define your own top level spell checking default.

**Comment by CaiJinKen** [comment]
I don`t think it`s a good idea to use spell checker in brackets match plugin. I mean the plugin just display brackets with rainbow maybe better, keep it simple, or provide some options to user.
Here is part of my .vimrc configuration maybe useful to others:
```vim
" set spell spelllang=en_us                       " turn of spell check

let g:rainbow_active = 1                            " active rainbow
nmap <leader>p :RainbowToggle<cr>     " set key map
```
